server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/EmailSubmissionSetMethod.scala

```diff
@@ -337,6 +338,24 @@ class EmailSubmissionSetMethod @Inject()(serializer: EmailSubmissionSetSerialize
       Failure(new IllegalArgumentException("Invalid delayed time!"))
     }
 
+  def validateMimeMessages(mimeMessage: MimeMessage) : SMono[MimeMessage] = validateMailAddressHeaderMimeMessage(mimeMessage)
+  private def validateMailAddressHeaderMimeMessage(mimeMessage: MimeMessage): SMono[MimeMessage] =
+    SFlux.fromIterable(Map("to" -> Option(mimeMessage.getRecipients(RecipientType.TO)).toList.flatten,
+        "cc" -> Option(mimeMessage.getRecipients(RecipientType.CC)).toList.flatten,
+        "bcc" -> Option(mimeMessage.getRecipients(RecipientType.BCC)).toList.flatten,
+        "from" -> Option(mimeMessage.getFrom).toList.flatten,
+        "sender" -> Option(mimeMessage.getSender).toList,
+        "replyTo" -> Option(mimeMessage.getReplyTo).toList.flatten))
+      .doOnNext { case (headerName, addresses) => (headerName, addresses.foreach(address => validateMailAddress(headerName, address))) }
+      .`then`()
+      .`then`(SMono.just(mimeMessage))
+
+  private def validateMailAddress(headName: String, address: Address): MailAddress =
+    Try(new MailAddress(address.toString)) match {
```

Comment:
```
Try(new MailAddress(address.getAddress)) match {
```

Reply:
class `Address` do not have `.getAddress` method